src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -804,4 +820,209 @@ public Owner syncOwnerLastContentUpdate(Owner owner) {
         return this.ownerCurator.merge(owner);
     }
 
+    /**
+     * Retrieves an existing {@link AnonymousContentAccessCertificate} for the {@link AnonymousCloudConsumer}
+     * or creates a new certificate if the existing certificate is expired or does not exist.
+     *
+     * @param consumer
+     *  the anonymous consumer to retrieve or create an {@link AnonymousContentAccessCertificate} for
+     *
+     * @throws IllegalArgumentException
+     *  if the provided anonymous cloud consumer is null
+     *
+     * @return
+     *  the certificate or null if unable to retrieve or create a certificate
+     */
+    @Transactional
+    public AnonymousContentAccessCertificate getCertificate(AnonymousCloudConsumer consumer) {
+        if (consumer == null) {
+            throw new IllegalArgumentException("anonymous cloud consumer is null");
+        }
+
+        try {
+            AnonymousContentAccessCertificate cert = consumer.getContentAccessCert();
+            Date now = new Date();
+            // Generate a new certificate if one does not exist or the existing certificate is expired
+            if (cert == null || cert.getSerial() == null || cert.getSerial().getExpiration().before(now)) {
+                cert = createAnonContentAccessCertificate(consumer);
+            }
+
+            if (cert == null) {
+                return null;
+            }
+
+            AnonymousContentAccessCertificate wrappedCert = new AnonymousContentAccessCertificate();
+            wrappedCert.setCert(cert.getCert() + cert.getContent());
+            wrappedCert.setCreated(cert.getCreated());
+            wrappedCert.setUpdated(cert.getUpdated());
+            wrappedCert.setId(cert.getId());
+            wrappedCert.setAnonymousCloudConsumer(cert.getAnonymousCloudConsumer());
+            wrappedCert.setKey(cert.getKey());
+            wrappedCert.setSerial(cert.getSerial());
+
+            return wrappedCert;
+        }
+        catch (Exception e) {
+            String msg = "Unexpected exception occurred while fetching certificate for anonymous" +
+                "cloud consumer: {}";
+            log.error(msg, consumer, e);
+        }
```

Comment:
We should limit the scope of try blocks to places that actually throw exceptions. Wrapping whole method body can get confusing as to why is the scope so wide.

Also something I noticed. The flow currently does `if null -> log generic error -> return null` on each step in the flow. Wouldn't exception make more sense? The code is not recovering from this anyway.

It would also allow resources to return errors to indicate what went wrong.

Reply:
This is a good point. We really should be using Exceptions instead of logging and then returning null. Updated to throw exceptions instead of returning null